### PR TITLE
[#150685848] Report CIMB - implement orderCapabilities on equity ticket

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItPreviewTradeOrderDetails+Helpers.swift [hunk 1]
```diff
@@ -1,31 +1,122 @@
...
+        default:
+            return "unknown"
+        }
+    }
+    
+    public init?(rawValue: RawValue) {
+        self.init(value: rawValue)
+    }
+    
+    public init(value: RawValue?) {
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItPreviewTradeOrderDetails+Helpers.swift]: Do these work for Obj-C users?

**Comment by jsom** [comment]
How will this work for anyone that has integrated their own custom UI and are trying to use the enums to set values?

File: TradeItIosTicketSDK2/TradeItPreviewTradeOrderDetails+Helpers.swift (hunk 1 above)
Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItPreviewTradeOrderDetails+Helpers.swift]: No, in Objective C, you can use only the enum `TradeItOrderActionBuy`, ...

**Comment by guillaumedebavelaere** [comment]
What do you mean ? 
Example to get the label:
```
self.orderCapabilities?.labelFor(field: .priceTypes, value: self.order.type.rawValue)
```

**Comment by jsom** [comment]
Some partners construct their own order objects with:

```swift
let order = TradeItOrder()
order.type = .limit
```

It seems like with this change we will no longer support this. I don't think it's a blocker, but we should do some combination of:

1. Put this on a new minor version
2. Contact partners that may do this to tell them that they need to upgrade and instruct them on how to use the order caps object

**Comment by guillaumedebavelaere** [comment]
```
let order = TradeItOrder()
order.type = .limit
```

This is still working. But yes, if they implement their own ticket, we should write a documentation on how do use the order capabilities.

**Comment by jsom** [comment]
Oh ok - so if they pass a limit order to the order object and submit it to a broker that doesn't support it, it will just fail server side?

**Comment by guillaumedebavelaere** [comment]
Yes

**Comment by jsom** [comment]
@guillaumedebavelaere can you rebase and merge

**Comment by guillaumedebavelaere** [comment]
@jsom @mitochondrion rebased. Could you review the last commit, I had to fix conflicts on some code you changed.

File: TradeItIosTicketSDK2/Constants.swift [hunk 2]
```diff
@@ -67,6 +67,7 @@ public struct TradeItNotification {
         case trading
         case preview
         case submitted
+        case selectActionType
```
Comment (jsom) [inline TradeItIosTicketSDK2/Constants.swift]: Can you add this to the objective-c mapping above in `TradeItNotificationConstants`

Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/Constants.swift]: @jsom done

**Pull request merged**